core/src/consensus.rs

```diff
@@ -264,6 +264,11 @@ impl Tower {
     }
 
     pub fn has_voted(&self, slot: u64) -> bool {
+        // slot 0 is special since it is a known root but it also needs to be voted on
+        if slot > 0 && Some(slot) == self.root() {
```

Comment:
ick

Reply:
I considered the other "fix" of updating bank_forka to use an option but it is eithe initialized via genesis or a snapshot so it's always got a known root. The only problem is is 0 is weird. Alternatively, I can close this pr since it's not a major issue. The "is_recent" check in "is_locked_out" will prevent repeated votes on the root Bank anyway.